### PR TITLE
[PE-4556] Fixed behavior of sidenav single level menu items

### PR DESCRIPTION
**Pull request by janibolkvadze**
This fix is necessary to improve the behavior of Sidenav menu items with no associated Drawer.

@dani-cl-madrid @jllr @mattnickles 

https://ctl.atlassian.net/browse/PE-4556

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://dc2f78beaf493e5a113d7f31173bd7cd01059516-chi-pr.enterprise.ctl.io" target="_blank"> https://dc2f78beaf493e5a113d7f31173bd7cd01059516-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/800/ for more details.

**Comment by mattnickles** [comment]
@dani-cl-madrid can you review when you have a second?

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://4b183ef504240bd28a35681158933e7d9c0d2c70-chi-pr.enterprise.ctl.io" target="_blank"> https://4b183ef504240bd28a35681158933e7d9c0d2c70-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/804/ for more details.

File: src/chi/javascript/components/sidenav.js [hunk 1]
```diff
@@ -96,6 +98,46 @@ class Sidenav extends Component {
...
+  singleLevelMenuItems() {
+    this._elem
+    .querySelectorAll(`nav ul.${LINKLIST_CLASS} li`)
+    .forEach(singleLevelMenuItem => {
+      const menuElementLink = singleLevelMenuItem.querySelector('a').getAttribute('href');
+
+      if (!this._elem.querySelector(`.chi-drawer${menuElementLink}`)) {
+        let menuItemToActivate;
+
+        singleLevelMenuItem.addEventListener("click", (e) => {
```
Comment (dani-cl-madrid) [inline src/chi/javascript/components/sidenav.js]: Are this event listeners removed on the disposal of this component?

File: src/chi/javascript/components/sidenav.js [hunk 2]
```diff
@@ -96,6 +98,46 @@ class Sidenav extends Component {
...
+          const activeMenuItem = this.getActiveMenuItem();
+
+          for (
+            let cur = e.target;
+            cur && !cur.classList.contains(LINKLIST_CLASS);
+            cur = cur.parentNode
+          ) {
+            menuItemToActivate = cur;
```
Comment (dani-cl-madrid) [inline src/chi/javascript/components/sidenav.js]: Wouldn't it better to check if cur is the element you are looking for? So then, you'll be able check if it existed, and if not, alert the developer that he didn't build the component properly.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://5936d70691094eb9fb0fd937ce07409b7e29e917-chi-pr.enterprise.ctl.io" target="_blank"> https://5936d70691094eb9fb0fd937ce07409b7e29e917-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/815/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-kad3o9, https://ops.elasticbox.com/services/instances/i-2cl2jv, https://ops.elasticbox.com/services/instances/i-2o5d30

**Pull request merged**